pwa/src/components/translations/translationTable.tsx

```diff
@@ -20,19 +13,17 @@ export default function TranslationTable({ tableName }) {
   const API: APIService = React.useContext(APIContext);
   const [_, setAlert] = React.useContext(AlertContext);
   const [__, setHeader] = React.useContext(HeaderContext);
-
-  React.useEffect(() => {
-    setHeader({ title: "Translations", subText: "An overview of your translation objects" });
-  }, [setHeader]);
-
+  
   React.useEffect(() => {
-    handleSetDocumentation(); // we added this
+    setHeader({title: 'Translations', subText: 'An overview of your translations'});
   });
```

Comment:
This goes for **everywhere** you're setting `setHeader()`:

1. By setting `setHeader` you're triggering a re-render every time and are therefore getting max-depth errors (huge performance issue, we definitely have to fix this)

To fix this, we can add the `setHeader` setter to the `useEffect` dependency array. This means: only re-run this block of code when (most likely if) `setHeader` changes, like so:

```
React.useEffect(() => {
  setHeader({ title: "Translation tables", subText: "An overview of your translation tables" });
}, [setHeader]); // this array is the dependency array
```

2. You're setting the header `setHeader()` in components **and** pages, which means you're setting them double (I'm not sure if you're doing this everywhere but for sure in the following example). I think it's best to set the headers in `pages` and not in `components` (and I think that's how Sarai meant it to be implemented when building it). Example of where I think you're duplicating the set header code (I assume you can just remove it from the component):

`TranslationTable` component:
```
React.useEffect(() => {
  setHeader({ title: "Translations", subText: "An overview of your translations" });
}, [setHeader]);
```

`Translations` page:
```
React.useEffect(() => {
  setHeader({ title: "Translations", subText: "An overview of your translations" });
}, [setHeader]);
```